shoplist/src/main/java/com/netguru/codereview/ui/MainViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.netguru.codereview.ui
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.netguru.codereview.network.ShopListApiMock
+import com.netguru.codereview.network.ShopListRepository
+import com.netguru.codereview.network.model.ShopListItemResponse
+import com.netguru.codereview.network.model.ShopListResponse
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class MainViewModel : ViewModel() {
+
+    private val shopListRepository = ShopListRepository(ShopListApiMock())
+
+    val shopLists = MutableLiveData<List<Pair<ShopListResponse, List<ShopListItemResponse>>>>()
+    private val eventLiveData = MutableLiveData<String>()
+
+    init {
+        viewModelScope.launch {
```

Comment:
viewModelScope.launch by default running on main/ui thread, so you must run this logic on another thread. 

You can do that by run this scope on Dispatchers.IO 

```suggestion
viewModelScope.launch {
            withContext(Dispatchers.IO) {
```